src/main/java/org/apache/sysml/hops/AggBinaryOp.java

```diff
@@ -344,11 +344,30 @@ protected double computeOutputMemEstimate( long dim1, long dim2, long nnz )
 	protected double computeIntermediateMemEstimate( long dim1, long dim2, long nnz )
 	{
 		double ret = 0;
-		
+
+		if (DMLScript.USE_ACCELERATOR) {
+			// In GPU Mode, these are the intermediate estimates for matmult:
+			// Both dense or both sparse - no intermediate
```

Comment:
please avoid redundant (1st line) and contradictory (2nd line) comments; we're in a method that only computes the intermediate memory requirements so saying "no intermediate" is confusing.